pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BrokerResourceOnlineOfflineStateModelFactory.java

```diff
@@ -97,14 +98,14 @@ public void onBecomeOnlineFromOffline(Message message, NotificationContext conte
 
     @Transition(from = "ONLINE", to = "OFFLINE")
     public void onBecomeOfflineFromOnline(Message message, NotificationContext context) {
-      String tableNameWithType = message.getPartitionName();
-      LOGGER.info("Processing transition from ONLINE to OFFLINE for table: {}", tableNameWithType);
+      String physicalOrLogicalTable = message.getPartitionName();
+      LOGGER.info("Processing transition from ONLINE to OFFLINE for table: {}", physicalOrLogicalTable);
       try {
-        _routingManager.removeRouting(tableNameWithType);
-        _queryQuotaManager.dropTableQueryQuota(tableNameWithType);
+        _routingManager.removeRouting(physicalOrLogicalTable);
```

Comment:
Will this cause a log to be printed for logical table that there is no routing for it ? Can you add an if condition so that the log files are not misleading ?  Was this function always being called and we missed handling logical tables correctly here ?

Reply:
This was not missed, we created a separate issue for remove table handling. https://github.com/apache/pinot/issues/15835

For the logical table, when a table is deleted, it would print.

`LOGGER.warn("Routing does not exist for table: {}, skipping removing routing", tableNameWithType);`